### DISCUSSION TITLE
Feature request (?) extend ssh config keywords

### DISCUSSION THREAD

**Discussion by jjo93sa** [blinksh/blink#1586] [Ideas]
Hi, thanks for creating Blink. I try (for better or worse) to use the same ssh config file across all my devices. I also cannot access all my hosts externally, needing to jump through a proxy. I do this in the config file like this:

```
Host internal.my.org
   HostName internal.my.org
   IdentityFile ~/.ssh/internal

Host jump*
   HostName jump.my.org
   IdentityFile ~/.ssh/jump
   Port 22

# ProxyJump through jump.poshboy.net but not for jump, or hosts with
# DNS entries
Match !exec "host %h" host *.my.org,!jump*
   ProxyJump me@jump.my.org

Host *
   User me
```

Inside my network `"internal.my.org"` resolves to an RFC1918 address, and ssh connects to it directly. Outside my network, on the public internet, the call to`"host %h"` in the `Match` clause fails, which because it is negated, results in the connection using the specified ProxyJump. Thus, no matter whether I'm internal or external to my network I can use the same ssh command and everything automagically works.

I've tried this in Blink, but I get a warning, which is:

`ssh_config_parse_line: line 12: Unsupported Match keyword !exec, ignoring` 

Which makes it seem like `exec` isn't supported, but it seems to be supported within the libssh code. I've also tried removing the negation, but then I get the error:

`ssh_config_parse_line: line 12: Unsupported Match keyword exec, ignoring` 

Am I doing something wrong? Are there plans to support (negated) exec in Match clauses?

Thanks!

**Comment by jeanger** [comment]
+1

**Comment by carloscabanero** [comment]
We added the rest of the SSH Config flags a while ago. We won't be supporting Match clauses. The main reason is because that requires to launch a background process to capture the output. It is doable, but we cannot rely on the current subsystem to do that at the moment. This is very niche and unfortunately to survive we have other battles. 

I will mark as won't fix and move to Discussions. If enough people needs it, we will consider it.
